blog/_posts/2025-03-23-roofit-ad.md

```diff
@@ -0,0 +1,154 @@
+---
+ title: "Automatic differentiation in RooFit with Clad"
+ layout: archive
+ author: Jonas Rembser
+---
+
+Maybe you have heard that RooFit now supports [Automatic Differentiation (AD)](https://en.wikipedia.org/wiki/Automatic_differentiation) for faster minimization.
+This blogpost explains you the underlying technology, implementation, and interfaces.
+It also showcases some performance numbers from a Higgs analysis benchmark before concluding with an outlook on future developments.
+
+## Introduction to Automatic differentiation with Clad
+
+### Analytic derivatives with Clad
+
+Did you know that ROOT can do **Automatic Differentiation (AD)** of mathematical C++ functions using the built-in [Clad compiler plugin](https://github.com/vgvassilev/clad)?
+
+Here is a simple example on how to create an automatic derivative with Clad in ROOT:
+
+```c++
+#include <Math/CladDerivator.h>
+
+double f(double x, double y) { return x * x * y; }
+
+// Call clad to generate the derivative of f wrt x.
+auto f_dx = clad::differentiate(f, "x");
+
+// Execute the generated derivative function.
+std::cout << "f_dx(3, 4) = << " f_dx.execute(/*x=*/3, /*y=*/4) << std::endl;
+
+// Dump the generated derivative code to standard output.
+f_dx.dump();
+```
+
+The output will be:
+```txt
+f_dx(3, 4) = 24
+
+The code is:
+double f_darg0(double x, double y) {
+    double _d_x = 1;
+    double _d_y = 0;
+    double _t0 = x * x;
+    return (_d_x * x + x * _d_x) * y + _t0 * _d_y;
+}
+```
+
+As you can see from the example above, Clad implements AD based on *source code transformation*, generating gradient code that can then be optimized by the compiler. Theoretically, this will give you better performance compared to doing AD with libraries for tensor computations oriented towards machine learning.
```

Comment:
Better performance for what application?

Reply:
Any application hopefully :slightly_smiling_face: